### PR TITLE
update code for default provenance

### PR DESCRIPTION
**Pull request by AyushVerma2**
update code for default provenance

### CODE REVIEW

File: src/integration-test/java/connection_check/ConnectionChecker.java [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-        return isServerReachable();
-    }
-
-    /**
-     * Checks HTTP URL and returns <code>true</code> a connection can be established
-     * to the corresponding host and port
-     *
-     * @return boolean if endpoint is up within the timeout
-     */
-    public boolean isServerReachable() {
```
Comment (mikera) [inline src/integration-test/java/connection_check/ConnectionChecker.java]: What is replacing this functionality? Is it being used at all? Should it be available as a utility function?

File: src/integration-test/java/developerTC/AgentService.java [hunk 2]
```diff
@@ -30,17 +27,19 @@
...
     private static String username;
     private static String password;
 
 
+
+    private static RemoteAccount remoteAccount;
+
+
     static {
         Properties properties = getProperties();
```
Comment (mikera) [inline src/integration-test/java/developerTC/AgentService.java:43]: Why are we doing all of this? Can we not use the Status API to connect via URL rather than manually constructing a DDO?

File: src/integration-test/java/developerTC/AgentService.java [hunk 3]
```diff
@@ -155,32 +129,10 @@ public static RemoteAgent getRemoteAgent() {
         return surfer;
 
     }
-    public static boolean isServerReachable(String uri) {
```
Comment (mikera) [inline src/integration-test/java/developerTC/AgentService.java]: Can you test the Agent Status functionality here (DEP 11)?

File: src/integration-test/java/developerTC/TestAgentEndpointQuery_06.java [hunk 4]
```diff
@@ -1,77 +0,0 @@
...
-
-/**
- * As a developer working with Ocean,
- * I need to obtain service endpoints for an arbitrary Agent in the Ocean ecosystem
- */
-public class TestAgentEndpointQuery_06 {
-
-
-    @Test
-    public void testServiceEndPoint() {
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentEndpointQuery_06.java:27]: A lot of these changes / deletions see unrelated to the PR title (update for default provenance).

Should these perhaps be a separate PR?

File: src/integration-test/java/developerTC/TestAgentEndpointUpdate_05.java [hunk 5]
```diff
@@ -37,7 +39,7 @@ public void testServiceEndPoint() {
         assertNull(remoteAgent.getAuthEndpoint());
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentEndpointUpdate_05.java]: Why are we checking for a null auth endpoint? Is this expected?

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java [hunk 6]
```diff
@@ -1,84 +1,66 @@
...
-        //Should not allow to create the null DID ?
-        //getting the default Ocean instance
-        Resolver resolver=new LocalResolverImpl();
-        RemoteAgent remoteAgent = RemoteAgent.create(resolver, null);
-        //registering the  DID and DDO
+        Resolver resolver=DexResolver.create();
+        DID did = DID.createRandom();
+        resolver.registerDID(did,getAgentDDO());
 
-        assertThrows(IllegalArgumentException.class, () -> {
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentRegistration_04.java:75]: Shouldn't this fail? Negative tests are good in general, best to include stuff like this.

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java [hunk 7]
```diff
@@ -1,84 +1,66 @@
...
-            resolver.registerDID(null, ddoString);
-        });
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentRegistration_04.java]: Need to test that getting the DDO for the registered DID returns exactly the same DDO String

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java [hunk 8]
```diff
@@ -1,84 +1,66 @@
...
-        String ddoString = JSON.toPrettyString(ddo);
 
-        //Should not allow to create the null DID ?
-        //getting the default Ocean instance
-        Resolver resolver=new LocalResolverImpl();
-        RemoteAgent remoteAgent = RemoteAgent.create(resolver, null);
-        //registering the  DID and DDO
+        Resolver resolver=DexResolver.create();
+        DID did = DID.createRandom();
+        resolver.registerDID(did,getAgentDDO());
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentRegistration_04.java]: Does this automatically add the DID to the DDO as required by the standard?

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java [hunk 9]
```diff
@@ -1,84 +1,66 @@
...
 
+        Asset asset = MemoryAsset.createFromString("Asset from string");
+        Asset registeredAsset = remoteAgent.registerAsset(asset);
 
+        Asset assetFromAgent = remoteAgent.getAsset(asset.getAssetID());
 
+        assertEquals(remoteAgent.getDID(), did);
+        assertEquals(remoteAgent.getMetaEndpoint(),  AgentService.getSurferUrl()+"/api/v1/meta");
+        assertEquals(registeredAsset.getMetadataString(), assetFromAgent.getMetadataString());
+
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAgentRegistration_04.java]: Check content of downloaded asset?

File: src/integration-test/java/developerTC/TestAssetRegistration_08.java [hunk 10]
```diff
@@ -186,4 +189,48 @@ public void testGetDIDAllTypes() throws IOException {
...
+
+        Map<String,Object> additionalMeta = new HashMap<>();
+        additionalMeta.put("name","BMW data ");
+        DataAsset  resourceAsset = ResourceAsset.create("assets/test_content.json",additionalMeta);
+
+        //  include Provenace
+        resourceAsset= resourceAsset.includeProvenace();
+
+        Map<String, Object> provData = JSON.toMap(resourceAsset.getMetadata().get("provenance").toString());
+        assertTrue(provData.get("wasAssociatedWith").toString().contains("BMW data "));
```
Comment (mikera) [inline src/integration-test/java/developerTC/TestAssetRegistration_08.java]: Why do we expect the string to be in the wasAccociatedWith block? Can we have some better description of what this is testing?

File: src/main/java/sg/dex/starfish/DataAsset.java [hunk 11]
```diff
@@ -128,4 +129,25 @@ default DataAsset updateMeta(String newMetaData) {
...
     }
 
+    /**
+     * This method is to include the include provenace in the asset metadata.
+     * Create default provenance metadata for publishing an asset.
+     * This create a random activity ID for tracking the provenance.
+     *
+     * @return respective data asset sub class.
+     */
+    default DataAsset includeProvenace()  {
```
Comment (mikera) [inline src/main/java/sg/dex/starfish/DataAsset.java]: What is this meant to be doing? How do you create provenance for an asset without extra information being provided?

File: src/main/java/sg/dex/starfish/DataAsset.java [hunk 12]
```diff
@@ -128,4 +129,25 @@ default DataAsset updateMeta(String newMetaData) {
...
+
+        if (null != this.getMetadata().get(Constant.PROVENANCE) ) {
+            return this;
+
+        } else {
+            Map<String, Object> metaMap = this.getMetadata();
+            String  name  =null != metaMap.get("name") ?
```
Comment (mikera) [inline src/main/java/sg/dex/starfish/DataAsset.java]: Explain this logic.

File: src/main/java/sg/dex/starfish/DataAsset.java [hunk 13]
```diff
@@ -128,4 +129,25 @@ default DataAsset updateMeta(String newMetaData) {
...
+            metaMap.get("name").toString() :"Prov_data_of";
+            metaMap.put(Constant.PROVENANCE,   ProvUtil.createPublishProvenance(name));
```
Comment (mikera) [inline src/main/java/sg/dex/starfish/DataAsset.java]: Why are we calling createPublishProvenance with the name of the asset or the String "Prov_data_of". Neither appear to be an agent ID....

**Review by mikera — CHANGES_REQUESTED** [review]
This PR seems to contain a lot of stuff unrelated to default provenance. Suggest maybe splitting.

File: src/integration-test/java/developerTC/AgentService.java (hunk 2 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/AgentService.java:43]: Reverted the changes, will raise PR separately for a separate issue.

File: src/integration-test/java/developerTC/AgentService.java (hunk 3 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/AgentService.java]: Agent Status is not yet implemented, will do tomorrow

File: src/integration-test/java/developerTC/TestAgentEndpointUpdate_05.java (hunk 5 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/TestAgentEndpointUpdate_05.java]: Reverted the changes, will raise PR separately for a separate issue.

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java (hunk 8 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/TestAgentRegistration_04.java]: Reverted the changes, will raise PR separately for a separate issue.
Also registration is done by line resolver.registerDID(did,getAgentDDO());

File: src/integration-test/java/developerTC/TestAgentRegistration_04.java (hunk 6 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/TestAgentRegistration_04.java:75]: Reverted the changes, will raise PR separately for a separate issue.

File: src/integration-test/java/developerTC/TestAssetRegistration_08.java (hunk 10 above)
Comment (AyushVerma2) [reply src/integration-test/java/developerTC/TestAssetRegistration_08.java]: Reverted the changes, will raise PR separately for a separate issue.

File: src/main/java/sg/dex/starfish/DataAsset.java (hunk 12 above)
Comment (AyushVerma2) [reply src/main/java/sg/dex/starfish/DataAsset.java]: Reverted the changes, will raise PR separately for a separate issue.
